src/status_im/ui/screens/add_new/models.cljs

```diff
@@ -4,19 +4,16 @@
             [status-im.i18n :as i18n]
             [status-im.ui.screens.navigation :as navigation]
             [status-im.utils.universal-links.core :as universal-links]
-            [status-im.utils.fx :as fx]))
+            [status-im.utils.fx :as fx]
+            [status-im.browser.core :as browser]))
 
-(fx/defn process-qr-code
+(fx/defn handle-qr-code
   [cofx data]
   (if (spec/valid? :global/public-key data)
     (universal-links/handle-view-profile cofx data)
     (or (universal-links/handle-url cofx data)
-        {:utils/show-popup {:title      (i18n/label :t/unable-to-read-this-code)
-                            :content    (i18n/label :t/use-valid-qr-code {:data data})
-                            :on-dismiss #(re-frame/dispatch [:navigate-to-clean :home])}})))
-
-(fx/defn handle-qr-code
-  [cofx data]
-  (fx/merge cofx
-            (navigation/navigate-to-clean :home {})
-            (process-qr-code data)))
+        (if (clojure.string/includes? data ".")
```

Comment:
I think you can use something like that to check for URLs:
```clojure
(defn is-url [url] (try (and (js/URL. url) true) (catch js/Error e false)))
```

Reply:
hmm this seems to always return true

Reply:
hmm, I tried it in a repl and for me it returns `false` for, say, “aaa” and `true` for “http://google.com"...

Reply:
hmm I'm using figwheel for status-im, and returns true for anything. I did try in a web repl, and it does return `false` for something like "aaa". which repl did you use?

Reply:
I think this should be handled the same way a URL is handled when tapped on by a user in chat so I would like to see where in the code that renders a chat message as a link. for example, if a message contains "aaaa google.com" then the 'google.com' becomes underlined. I'm having trouble trying to find where this happens :/

Reply:
@yenda maybe you know?

Reply:
we have a `regx-url` in `constants.cljs` to check for urls